File: web/src/utils/escapeBackslashes.ts
import isPunctuation from "../utils/isPunctuation";

export default function escapeBackslashes(text: string): string {
	let result = "";
	for (let i = 0; i < text.length; i++) {
		let char = text[i];
		if (char === "\\" && isPunctuation(text.charCodeAt(i + 1))) {
			i++;
			char = text[i];
		}
		result += char;
	}
	return result;
}

File: dotnet/Allmark/Utils/EscapeBackslashes.cs
namespace Allmark;

public static partial class Utils
{
	public static string EscapeBackslashes(string text)
	{
		var result = "";
		for (int i = 0; i < text.Length; i++)
		{
			char c = text[i];
			if (c == '\\' && IsPunctuation((int)text[i + 1]))
			{
				i++;
				c = text[i];
			}
			result += c;
		}
		return result;
	}
}
